compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -1521,6 +1526,210 @@ LogicalResult ScaledMMAAttr::buildUnderlyingOperations(
+  int64_t k = getKSize(intrinsic) * getKbSize(intrinsic);
+  int64_t n = getNSize(intrinsic);
+
+  return amdgpu::ScaledMFMAOp::create(builder, loc, m, n, k, lhs, rhs, acc,
+                                      lhsScales, rhsScales, /*scalesIdxA=*/0,
+                                      /*scalesIdxB=*/0);
+}
+
+LogicalResult DataTiledScaledMMAAttr::buildUnderlyingOperations(
+    OpBuilder &builder, Location loc, ValueRange inputs, ValueRange outputs,
+    SmallVectorImpl<Value> &results) const {
+  // Validation. Similar to MMAAttr::buildMmaOperation.
+  if (inputs.size() != 4) {
+    return failure();
+  }
+  if (outputs.size() != 1) {
+    return failure();
+  }
+  SmallVector<VectorType> regTypes;
+  getDistributedTileTypes(regTypes);
+  if (!llvm::equal(regTypes,
+                   llvm::concat<Type>(inputs.getTypes(), outputs.getTypes()))) {
+    return failure();
+  }
+
+  // Prepare Lhs/Rhs/Acc operand slices to feed the intrinsic.
+  const unsigned lhsIdx = 0;
+  const unsigned rhsIdx = 1;
+  const unsigned lhsScalesIdx = 2;
+  const unsigned rhsScalesIdx = 3;
+  const unsigned accIdx = 4;
+  TileSwizzle lhsSwizzle = getSwizzle(*this, lhsIdx);
+  LDBG() << "DataTiledScaledMMAAttr::buildMmaOperation";
+  LDBG() << "    lhsSwizzle: " << lhsSwizzle;
+  SmallVector<Value> intrinsicsLhs =
+      distributeMmaFragmentToIntrinsics(builder, loc, inputs[0], lhsSwizzle);
+
+  TileSwizzle rhsSwizzle = getSwizzle(*this, rhsIdx);
+  LDBG() << "DataTiledScaledMMAAttr::buildMmaOperation";
+  LDBG() << "    rhsSwizzle: " << rhsSwizzle;
+  SmallVector<Value> intrinsicsRhs =
+      distributeMmaFragmentToIntrinsics(builder, loc, inputs[1], rhsSwizzle);
+
+  TileSwizzle lhsScalesSwizzle = getSwizzle(*this, lhsScalesIdx);
+  LDBG() << "DataTiledScaledMMAAttr::buildMmaOperation";
+  LDBG() << "    lhsScalesSwizzle: " << lhsScalesSwizzle;
+  SmallVector<Value> intrinsicsLhsScales = distributeMmaFragmentToIntrinsics(
+      builder, loc, inputs[2], lhsScalesSwizzle);
+
+  TileSwizzle rhsScalesSwizzle = getSwizzle(*this, rhsScalesIdx);
+  LDBG() << "DataTiledScaledMMAAttr::buildMmaOperation";
+  LDBG() << "    rhsScalesSwizzle: " << rhsScalesSwizzle;
+  SmallVector<Value> intrinsicsRhsScales = distributeMmaFragmentToIntrinsics(
+      builder, loc, inputs[3], rhsScalesSwizzle);
+
+  TileSwizzle accSwizzle = getSwizzle(*this, accIdx);
+  LDBG() << "DataTiledScaledMMAAttr::buildMmaOperation";
+  LDBG() << "    accSwizzle: " << accSwizzle;
+
+  SmallVector<Value> intrinsicsAcc =
+      distributeMmaFragmentToIntrinsics(builder, loc, outputs[0], accSwizzle);
+
+  ScaledMMAIntrinsic intrinsic = getIntrinsic();
+  auto intrinCType = cast<VectorType>(intrinsicsAcc.front().getType());
+
+  // Loop over the 3 unroll_{m,n,k} dimensions to create the intrinsics.
+  for (int mu = 0; mu < getIntrinsicsM(); ++mu) {
+    for (int nu = 0; nu < getIntrinsicsN(); ++nu) {
+      for (int ku = 0; ku < getIntrinsicsK(); ++ku) {
```

Comment:
Nitting about types again:
```suggestion
  for (int64_t mu = 0; mu < getIntrinsicsM(); ++mu) {
    for (int64_t nu = 0; nu < getIntrinsicsN(); ++nu) {
      for (int64_t ku = 0; ku < getIntrinsicsK(); ++ku) {
```